src/external/transaction.rs

```diff
@@ -94,6 +94,31 @@ pub fn sender() -> Vec<u8> {
     }
 }
 
+pub trait TransactionInfo {
+    fn arguments(&self) -> Vec<u8>;
+    fn ret(&self, _values: &[u8]);
+    fn sender(&self) -> Vec<u8>;
+}
+
+#[derive(Clone)]
+struct TransactionInfoImpl {}
+impl TransactionInfo for TransactionInfoImpl {
+    fn arguments(&self) -> Vec<u8> {
+        arguments()
+    }
+    fn ret(&self, _values: &[u8]) {
```

Comment:
I don't think "values" needs to start with an underscore here, since it is being used in the function.